server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -693,6 +712,7 @@ public void run() {
                     "Compaction in progress, read %d of %d input entries ( %s %s ), written %d entries",
                     info.getEntriesRead(), inputEntries, percentComplete, "%",
                     info.getEntriesWritten());
+                watcher.run();
```

Comment:
Not sure if this can happen, but I think there may be a case where stuck compaction stats are not cleaned up because of how this is called.  It seems like its only called periodocally when a compaction is running.  If a compaction that was stuck starts again, but watcher.run() is not called because of timing issues (like the previously stuck compaction quickly completes) then it would continue to look stuck.

Reply:
In bec830ff3d0dc606864e5172fc463f80224f5ca7 I call `watcher.run()` after the compaction has finished to clear out the state.